internal/database/repository/queue.go

```diff
@@ -161,6 +163,81 @@ func (s *PostgresRepository) AddQueues(ctx context.Context, parentId *string, qu
 	return nil
 }
 
+// UpdateQueue updates the queue based on the queue_name and partition.
+// If the queue has children, the function will recursively update them.
+// If provided child queue does not exist, the function will add it.
+// The function returns an error if the update operation fails.
+func (s *PostgresRepository) UpdateQueue(ctx context.Context, queue *dao.PartitionQueueDAOInfo) error {
```

Comment:
Should calls like this be wrapped in a transaction?
cc @dejanzele

Reply:
I haven't given it much thought, it is a good question.

If there are 10 queues and it fails in the middle, should we:
1. revert all changes using transactions
2. stop at 5th and retry same query (in case of network issues or solvable issues)
3. finish 9 and retry the failed (and retry in case of network issues or solvable issues)

Might be worthwhile opening an Issue or Discussion